lib/fwupd_mock_service.dart

```diff
@@ -0,0 +1,171 @@
+import 'dart:io';
+
+import 'package:fwupd/fwupd.dart';
+import 'package:ubuntu_logger/ubuntu_logger.dart';
+import 'package:yaml/yaml.dart';
+
+import 'fwupd_service.dart';
+
+final log = Logger('fwupd_service');
+
+// Mainly no-op methods for testing
+class FwupdMockService extends FwupdService {
+  String? simulateYamlFilePath;
```

Comment:
Any reason not to put this in the constructor, i.e. `FwupdMockService({this.simulateYamlFilePath})`?